openedx/features/course_duration_limits/access.py

```diff
@@ -43,25 +51,32 @@ def get_user_course_expiration_date(user, course):
     """
     Return course expiration date for given user course pair.
     Return None if the course does not expire.
+    Defaults to MIN_DURATION.
     """
-    # TODO: Update business logic based on REV-531
+
+    access_duration = MIN_DURATION
+
     CourseEnrollment = apps.get_model('student.CourseEnrollment')
     enrollment = CourseEnrollment.get_enrollment(user, course.id)
-    if enrollment is None or enrollment.mode == 'verified':
+    if enrollment is None or enrollment.mode != 'audit':
         return None
 
     try:
         start_date = enrollment.schedule.start
     except CourseEnrollment.schedule.RelatedObjectDoesNotExist:
         start_date = max(enrollment.created, course.start)
 
-    access_duration = timedelta(weeks=8)
-    if hasattr(course, 'pacing') and course.pacing == 'instructor':
-        if course.end and course.start:
-            access_duration = course.end - course.start
+    if course.self_paced:
+        discovery_course_details = get_course_run_details(course.id, ['weeks_to_complete'])
```

Comment:
I believe this hits a cache before a request goes out to course-discovery?  
If not, it should as otherwise it would probably take down discovery with the increased load